lib/cloud_controller/diego/desire_app_handler.rb

```diff
@@ -12,6 +12,8 @@ def create_or_update_app(process, client)
               if e.name == 'RunnerError' && e.message['the requested resource already exists']
                 existing_lrp = client.get_app(process)
                 client.update_app(process, existing_lrp)
+              elsif e.name == 'RunnerError' && e.message["'root' user not permitted"]
```

Comment:
I know this is continuing an existing pattern, but this is pretty brittle to error message changes.

What about something like:
1. client/lifecycle raises a custom error type
2. handler catches that error type and re-raises API error